src/cluster.c

```diff
@@ -4904,16 +5091,23 @@ int verifyClusterConfigWithData(void) {
          * assigned to this slot. Fix this condition. */
 
         update_config++;
-        /* Case A: slot is unassigned. Take responsibility for it. */
+        /* slot is unassigned. Take responsibility for it. */
         if (server.cluster->slots[j] == NULL) {
             serverLog(LL_NOTICE, "I have keys for unassigned slot %d. "
                                     "Taking responsibility for it.",j);
             clusterAddSlot(myself,j);
-        } else {
-            serverLog(LL_NOTICE, "I have keys for slot %d, but the slot is "
-                                    "assigned to another node. "
-                                    "Setting it to importing state.",j);
-            server.cluster->importing_slots_from[j] = server.cluster->slots[j];
+        } else if (server.cluster->importing_slots_from[j] != server.cluster->slots[j]) {
+            if (server.cluster->importing_slots_from[j] == NULL) {
+                serverLog(LL_WARNING, "I have keys for slot %d, but the slot is "
+                                      "assigned to another node. Deleting keys in the slot.", j);
+            } else {
+                serverLog(LL_WARNING, "I am importing keys from node %.40s to slot %d, "
+                                      "but the slot is now owned by node %.40s. Deleting keys in the slot",
+                                      server.cluster->importing_slots_from[j]->name,
+                                      j,
+                                      server.cluster->slots[j]->name);
```

Comment:
```suggestion
                serverLog(LL_WARNING, "I am importing keys from node %.40s (%s) to slot %d, "
                                      "but the slot is now owned by node %.40s (%s). Deleting keys in the slot",
                                      server.cluster->importing_slots_from[j]->name, server.cluster->importing_slots_from[j]-> human_nodename
                                      j,
                                      server.cluster->slots[j]->name, server.cluster->slots[j]->human_nodename);
```
Human nodenames were added in the interim, which is supposed to give better log visibility by reporting a logical identifier instead of just the GUID for the node. If you can, please add it wherever it would be useful to have more information.